ci/environment-dev.yaml

```diff
@@ -13,3 +13,4 @@ dependencies:
   - pytz
   - setuptools>=24.2.0
   - sphinx
+  - sphinxcontrib-spelling
```

Comment:
I think we should rather put it in the requirements-pip/conda-optional.txt

Reply:
@jreback, you said it should be in `requirements_dev.txt`, may be you want to explain why? I also thought the optional files made more sense, but you two surely know better than me.

Reply:
requirements_dev are for pure dev things
but i think things have gotten confused

Reply:
OK, let's discuss that in another one then